### PR TITLE
enable kql extension install

### PR DESCRIPTION
**Pull request by NarineM**
Install and run KqlExtension proc only in 3P mode and non aarch64 platforms.

### CODE REVIEW

File: AzureMonitorAgent/agent.py [hunk 1]
```diff
@@ -1717,6 +1720,22 @@ def get_ssl_cert_info(operation):
 
     log_and_exit(operation, GenericErrorCode, 'Unable to determine values for SSL_CERT_DIR or SSL_CERT_FILE')
 
+def copy_kqlextension_binaries():
+    kqlextension_bin_local_path = os.getcwd() + "/KqlExtensionBin/"
```
Comment (simathih) [inline AzureMonitorAgent/agent.py:1724]: where are we adding contents to KqlExtensionBin? is that a separate PR?

Comment (NarineM) [reply AzureMonitorAgent/agent.py:1724]: yes, that is separate PR in tux repo to populate the content of the folder. This PR is for installing the folder content as part of extension install.

Comment (NarineM) [reply AzureMonitorAgent/agent.py:1744]: PR to populate kql bin: https://msazure.visualstudio.com/One/_git/Compute-Runtime-Tux/pullrequest/10283834

File: AzureMonitorAgent/agent.py [hunk 2]
```diff
@@ -595,6 +598,26 @@ def enable():
...
+    if check_dotnet != 0:
+        print(".NET 7.0 is not installed. Please install .NET 7.0 if you are using Kql transformation. See more here https://learn.microsoft.com/en-us/dotnet/core/install/linux")
+        #ensure kql extension service is not running. do not block if it fails
+        kql_exit_code, disable_output = run_command_and_log(get_service_command("azuremonitor-kqlextension", "stop", "disable"))
+        if kql_exit_code != 0:
+            status_command = get_service_command("azuremonitor-kqlextension", "status")
+            kql_exit_code, status_output = run_command_and_log(status_command)
+    else:
+        if "7.0" in dotnetcmd_output:
+            print("Found .NET 7.0 installed.")
```
Comment (NarineM) [inline AzureMonitorAgent/agent.py:612]: need to start kql extension here

Comment (AaronRM) [reply AzureMonitorAgent/agent.py:612]: Reminder to please start + enable. If we only start (but don't also enable), then the system will not start the service up after a reboot. We should typically be doing start/enable and stop/disable together.

File: AzureMonitorAgent/agent.py [hunk 3]
```diff
@@ -595,6 +599,31 @@ def enable():
                 output += "Output of '{0}':\n{1}".format(status_command, status_output)
                 return exit_code, output
 
+    # check if .NET is installed to start Kql extension process
+    if platform.machine() != 'aarch64':
+        check_dotnet, dotnetcmd_output = run_command_and_log("dotnet --list-runtimes",log_cmd=False)
+        if check_dotnet != 0:
+            print(".NET 7.0 is not installed. Please install .NET 7.0 if you are using Kql transformation. See more here https://learn.microsoft.com/en-us/dotnet/core/install/linux")
```
[the remark this answers is not in the document]
Comment (NarineM) [reply AzureMonitorAgent/agent.py]: updated.

**Review by simathih — APPROVED** [review]

File: AzureMonitorAgent/agent.py [hunk 4]
```diff
@@ -598,6 +598,26 @@ def enable():
...
 
+    # check if .NET is installed
+    check_dotnet, dotnetcmd_output = run_command_and_log("dotnet --list-runtimes",log_cmd=False)
+    if check_dotnet != 0:
+        print(".NET 7.0 is not installed. Please install .NET 7.0 if you are using Kql transformation. See more here https://learn.microsoft.com/en-us/dotnet/core/install/linux")
+        #ensure kql extension service is not running. do not block if it fails
+        kql_exit_code, disable_output = run_command_and_log(get_service_command("azuremonitor-kqlextension", "stop", "disable"))
+        if kql_exit_code != 0:
+            status_command = get_service_command("azuremonitor-kqlextension", "status")
+            kql_exit_code, status_output = run_command_and_log(status_command)
```
[the remark this answers is not in the document]
Comment (NarineM) [reply AzureMonitorAgent/agent.py]: logging this now. Just noticed some paths that have minor impact on extension do not get logged but the methods are using this style probably to consider adding these in the future. I have updated my PR though to log command output from all the paths even optional. Might consider removing it in the future if itis too verbose.

**Pull request merged**